From 319ec1563588aaae55d2ff075217db38e238e8cf Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Mon, 14 Oct 2024 16:45:15 -0700
Subject: [PATCH 2/3] ui/gtk: Always calculate scale factors

x and y scale factors should always be calculated
in gd_egl_draw.

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 ui/gtk-egl.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/ui/gtk-egl.c b/ui/gtk-egl.c
index af7e032d55..6e9cd241e5 100644
--- a/ui/gtk-egl.c
+++ b/ui/gtk-egl.c
@@ -80,6 +80,9 @@ void gd_egl_draw(VirtualConsole *vc)
     ww = gdk_window_get_width(window) * ws;
     wh = gdk_window_get_height(window) * ws;
 
+    vc->gfx.scale_x = (double)ww / surface_width(vc->gfx.ds);
+    vc->gfx.scale_y = (double)wh / surface_height(vc->gfx.ds);
+
     eglMakeCurrent(qemu_egl_display, vc->gfx.esurface,
                    vc->gfx.esurface, vc->gfx.ectx);
 
@@ -93,9 +96,6 @@ void gd_egl_draw(VirtualConsole *vc)
             }
         }
 #endif
-        vc->gfx.scale_x = (double)ww / surface_width(vc->gfx.ds);
-        vc->gfx.scale_y = (double)wh / surface_height(vc->gfx.ds);
-
         if (!dmabuf || (dmabuf && dmabuf->render_sync)) {
             gd_egl_scanout_flush(&vc->gfx.dcl, 0, 0, vc->gfx.w, vc->gfx.h);
         }
@@ -114,10 +114,7 @@ void gd_egl_draw(VirtualConsole *vc)
         eglSwapBuffers(qemu_egl_display, vc->gfx.esurface);
 
         gd_gl_count_frame(&vc->gfx.dcl, false);
-        vc->gfx.scale_x = (double)ww / surface_width(vc->gfx.ds);
-        vc->gfx.scale_y = (double)wh / surface_height(vc->gfx.ds);
     }
-
     glFlush();
 }
 
-- 
2.43.0

